### PR TITLE
fix for WICED HTTP server only reading one packet

### PR DESCRIPTION
**Pull request by m-mcgowan**
fix for issue #680.  When content length is specified, ensure subsequent packets are read as necessary.

Known issues:
- this probably doesn't work with chunked transfer across multiple packets. I'm hoping that this doesn't arise in practice. 

To apply this change:

- update to 0.4.6 if not already done
- cd modules/photon/system-part2
- make PLATFORM_ID=6 all program-dfu

### CODE REVIEW

**Comment by m-mcgowan** [comment]
Tested by using a python HTTP softAP client. Also tested using a python script that sents 2 packets. Old code didn't return an error, while the new code does (decryption error since the pwd isn't encrypted.) Community testing underway.

**Comment by indraastra** [comment]
Thanks, this seems to be working! On the client side, setting the [content-type appropriately](https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS#Simple_requests) is required to prevent an OPTIONS request (that the server never replies to) preceding the POST, but once this is done, the server handles the POST like a champ!

**Comment by m-mcgowan** [comment]
Great to hear. For anyone else following a long, I posted the system-part2.bin image in the github issue https://github.com/spark/firmware/issues/680

File: hal/src/photon/softap.cpp [hunk 1]
```diff
@@ -901,16 +902,121 @@ static void tcp_stream_reader(Reader& r, wiced_tcp_stream_t* stream) {
...
+        if (result)
+            return -1;
+
+        body->user = packet;         // ensure we clean up the packet
+        uint16_t packet_length;
+        uint16_t available_data_length;
+        uint32_t offset =0;
+        uint8_t* data;
+
+        if (length>available_data_length)
```
Comment (pingdynasty) [inline hal/src/photon/softap.cpp]: @m-mcgowan comparing uninitialised variable `available_data_length`

Comment (m-mcgowan) [inline hal/src/photon/softap.cpp]: Thanks!

**Comment by pingdynasty** [comment]
any chance of posting the `http_server.c` patch, rather than just the binary?

**Pull request merged**